src/backends/torch/native/templates/nbeats.cc

```diff
@@ -44,7 +68,24 @@ namespace dd
     torch::Tensor forecast = tffc.mm(_fS.to(_device));
     return std::make_tuple(
         backcast.reshape({ backcast.size(0), _backcast_length, _data_size }),
-        forecast.reshape({ forecast.size(0), _forecast_length, _data_size }));
+        forecast.reshape({ forecast.size(0), _backcast_length, _data_size }));
```

Comment:
for now, backcast_length and forecast_length are forced to have the same value, this only because forward/extract return a tensor and so I have to stack backcast and forecast values into one tensor, and so they need to have the same dimension along time axis. This will be cleanup up as soon as I will be able to change everything to return an IValue and amke the corresponding changes / checks in torchlib.cc

Reply:
Ok